### PR TITLE
feat(stack): Allow sidecars to be marked as non-essential

### PR DESCRIPTION
**Pull request by tam0ri**
Now we can specify essential parameter for sidecars in service manifest.

Addresses #2066

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/deploy/cloudformation/stack/transformers_test.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ func Test_convertSidecar(t *testing.T) {
 	mockImage := aws.String("mockImage")
 	mockMap := map[string]string{"foo": "bar"}
 	mockCredsParam := aws.String("mockCredsParam")
+	mockEssential := aws.Bool(false)
```
Comment (efekarakus) [inline internal/pkg/deploy/cloudformation/stack/transformers_test.go]: nit: would you mind updating the testCases to take an `inEssential` field instead and in the `wanted` we can ensure that exact value is applied. Right now we don't have a scenario that tests if the conversion happens on `essential: true`

**Review by efekarakus — COMMENTED** [review]
Looks great! Thank you so much for the contribution 🙏 

Just a tiny request for updating the unit test

File: internal/pkg/deploy/cloudformation/stack/transformers_test.go (hunk 1 above)
Comment (tam0ri) [reply internal/pkg/deploy/cloudformation/stack/transformers_test.go]: I just fixed it!

**Review by efekarakus — APPROVED** [review]

**Comment by bvtujo** [comment]
This looks great! Thank you so much for this change.

**Review by bvtujo — APPROVED** [review]

**Pull request merged**